miio/viomivacuum.py

```diff
@@ -268,124 +347,340 @@ def has_map(self) -> bool:
 
     @property
     def has_new_map(self) -> bool:
-        """TODO: unknown"""
+        """True if the device has scanned a new map (like a new floor)."""
         return bool(self.data["has_newmap"])
 
     @property
     def mop_mode(self) -> ViomiMode:
-        """Whether mopping is enabled and if so which mode
+        """Whether mopping is enabled and if so which mode"""
+        return ViomiMode(self.data["is_mop"])
+
+    @property
+    def current_map_id(self) -> float:
+        """Current map id."""
+        return self.data["cur_mapid"]
+
+    @property
+    def hw_info(self) -> str:
+        """Hardware info."""
+        return self.data["hw_info"]
 
-        TODO: is this really the same as mode?
+    @property
+    def charging(self) -> bool:
+        """Battery is charging or not.
+
+        is_charge is 1 when the battery is not charging
+        is_charge is 0 when the device is charging
+        Note: When the battery is at 100% is_charge is 1
+
+        Return:
+        - True if the battery is charging
+        - False if the battery is NOT charging
         """
-        return ViomiMode(self.data["is_mop"])
+        return not (bool(self.data["is_charge"]))
+
+    @property
+    def working(self) -> bool:
+        """Device is working or not.
```

Comment:
See my suggestion above.

Also, could we rename this to `is_on` for consistency?

Reply:
fixed